src/components/PostPicker/index.js

```diff
@@ -0,0 +1,389 @@
+		] );
+	};
+
+	const reorderDown = ( postId ) => {
+		const index = values.indexOf( postId );
+
+		// Item not found or already at the last position
+		if ( index === -1 || index === values.length - 1 ) {
+			return;
+		}
+
+		onChange( [
+			...values.slice( 0, index ),
+			values[ index + 1 ],
+			values[ index ],
+			...values.slice( index + 2 ),
+		] );
+	};
+
+	return (
+		<div style={ {
+			// Overflow at top.
+			marginTop: -24,
+			paddingTop: 24,
+			// Offset to display checkbox focus outline.
+			paddingLeft: 4,
+			marginLeft: -4,
+		} }>
+			{
+				( isResolving && <Spinner /> )
+					|| ( queriedPosts.length < 1 &&
+						<Notice
+							isDismissible={ false }
+						>
+							{ __( 'No results found' ) }
+						</Notice>
+					)
+					|| (
+						queriedPosts.map( post => (
+							<div
+								style={ {
+									display: 'grid',
+									gridTemplateColumns: '1fr auto',
+									marginRight: -2,
+									paddingRight: 2,
+								} }
+							>
+								<CheckboxControl
+									key={ post.id }
+									checked={ values.includes( post.id ) }
+									label={ post.title?.rendered || __( '(No title)', 'block-editor-components' ) }
+									onChange={ ( checked ) => {
+										if ( checked ) {
+											onChange( [ ...values, post.id ] );
+										} else {
+											onChange( values.filter( value => ( value !== post.id ) ) );
+										}
+									} }
+								/>
+								{ isSortable && (
+									<ButtonGroup>
+										<Button
+											icon={ 'arrow-up-alt2' }
+											iconSize={ 12 }
+											isSmall
+											label={ __( 'Move up', 'block-editor-components' ) }
+											variant="secondary"
+											onClick={ () => reorderUp( post.id ) }
+										/>
+										<Button
+											icon={ 'arrow-down-alt2' }
+											iconSize={ 12 }
+											isSmall
+											label={ __( 'Move down', 'block-editor-components' ) }
+											variant="secondary"
+											onClick={ () => reorderDown( post.id ) }
+										/>
+									</ButtonGroup>
+								) }
+							</div>
+						) )
+					)
+			}
+		</div>
+	);
+}
+
+/**
+ * Post Picker Browse Post Panel.
+ *
+ * @param props
+ * @returns {ReactNode} Component
+ */
+function BrowsePanel( props ) {
+	const {
+		postType,
+		onChange,
+		values,
+		taxonomies,
+	} = props;
+
+	const [ search, setSearch ] = useState();
+
+	const taxObjects = useSelect( select => {
+		return taxonomies.map( taxonomy => select( 'core' ).getTaxonomy( taxonomy ) );
+	}, [ taxonomies ] );
+
+	const [ taxQueries, setTaxQueries ] = useState( [] );
+
+	/**
+	 * Helper function to update tax query for a taxonomy.
+	 *
+	 * Sets correct property value of rest_base.
+	 * Merges the new data with existing data.
+	 *
+	 * @param {*} taxonomy Taxonomy.
+	 * @param {*} newTerms New terms.
+	 */
+	const updateTaxQueryState = useCallback( ( taxonomy, newTerms ) => {
+		const taxObject = taxObjects.find( t => t && t.slug === taxonomy );
+
+		if ( taxObject ) {
```

Comment:
would this be easier to read if it returned early? I know it's an extra line of code but it'd reduce indenting